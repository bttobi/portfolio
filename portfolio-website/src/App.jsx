import './index.css';
import './components/Navbar'
import { Route, Routes, BrowserRouter } from "react-router-dom";
import Navbar from './components/Navbar';
import Home from './components/Home';
import Projects from './components/Projects';
import Contact from './components/Contact';

function App() {
  return (
    <BrowserRouter>
      <div className="App">
        <Navbar/>
        <Routes>
          <Route path='/' element={<Home/>}/>
          <Route path='projects' element={<Projects/>}/>
          <Route path='contact' element={<Contact/>}/>
        </Routes>
      </div>
    </BrowserRouter>
  )
}

export default App
