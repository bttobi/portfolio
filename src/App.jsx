import "./index.css";
import Navbar from "./components/Navbar";
import Home from "./components/pages/Home";
import Experience from "./components/pages/Experience";
import Contact from "./components/pages/Contact";
import { motion, AnimatePresence } from "framer-motion";
import { useEffect } from "react";
import isInView from "./components/functions/isInView";

function App() {
  useEffect(() => {
    document.addEventListener("scroll", isInView); //adding scrolling event
  });

  return (
    <AnimatePresence>
      <motion.div
        initial={{ opacity: 0 }}
        animate={{ opacity: 1 }}
        exit={{ opacity: 0 }}
        id="App"
        className="flex flex-col"
      >
        <Navbar />
        <Home />
        <Experience />
        <Contact />
      </motion.div>
    </AnimatePresence>
  );
}

export default App;
