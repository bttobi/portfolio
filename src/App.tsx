import { Navbar } from './components/ui_elements/navigation';
import { Pages, pagesArray } from './components/pages';
import './index.css';
import { useIntl } from 'react-intl';

const App = () => {
  const intl = useIntl();
  return (
    <main id="app" className="text-primary bg-bg">
      <Navbar items={pagesArray(intl)} />
      <Pages />
    </main>
  );
};

export default App;
