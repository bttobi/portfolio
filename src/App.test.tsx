import App from './App';
import { render } from './tests/unit_and_integration/test-utils';

const setup = () => render(<App />);

describe('Tests App component', () => {
  it('renders the App component', () => {
    const { container } = setup();
    expect(container).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
