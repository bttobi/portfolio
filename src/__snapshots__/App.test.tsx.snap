// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests App component > matches snapshot 1`] = `
<div>
  <main
    class="text-primary bg-bg"
    id="app"
  >
    <nav
      class="w-full h-16 bg-secondary flex justify-center align-center gap-10 fixed top-0"
    >
      <img
        alt="Bartosz TobiÅ„ski's logo"
        class="h-12 hidden md:flex"
        src="https://i.imgur.com/7oPtzQz.png"
      />
      <button
        aria-label="Go to home page"
        class="transition duration-150 hover:text-contrast"
        title="Go to home page"
      >
        home
      </button>
      <button
        aria-label="Go to about page"
        class="transition duration-150 hover:text-contrast"
        title="Go to about page"
      >
        about
      </button>
      <button
        aria-label="Go to contact page"
        class="transition duration-150 hover:text-contrast"
        title="Go to contact page"
      >
        contact
      </button>
      <button
        aria-label="Go to resume page"
        class="transition duration-150 hover:text-contrast"
        title="Go to resume page"
      >
        resume
      </button>
    </nav>
    <section
      class="h-screen w-full mt-16"
    >
      <div>
        Home
      </div>
    </section>
    <section
      class="h-screen w-full mt-16"
    >
      <div>
        About
      </div>
    </section>
    <section
      class="h-screen w-full mt-16"
    >
      <div>
        Contact
      </div>
    </section>
  </main>
</div>
`;
